app/server/social_auth.py

```diff
@@ -68,3 +70,64 @@ def fetch_azuread_permissions(strategy, details, user=None, is_new=False, *args,
     if user.is_superuser != is_superuser:
         user.is_superuser = is_superuser
         user.save()
+
+
+# noinspection PyUnusedLocal
+def fetch_okta_oauth2_permissions(strategy, details, user=None, is_new=False, *args, **kwargs):
+    org_url = getattr(settings, 'SOCIAL_AUTH_OKTA_OAUTH2_API_URL', '')
+    admin_group_name = getattr(settings, "OKTA_OAUTH2_ADMIN_GROUP_NAME", "")
+    if not user or not isinstance(kwargs['backend'], OktaOAuth2):
+        return
+
+    # OktaOpenIdConnect inherits `OktaOAuth2`, so we have to explicitly skip OAuth2 trying
+    # to fetch permissions when using OIDC backend.
+    if isinstance(kwargs['backend'], OktaOpenIdConnect):
+        return
+
+    response = requests.post(
+        url=f"{org_url}/v1/userinfo",
+        headers={
+            'Authorization': 'Bearer {}'.format(kwargs['response']['access_token']),
+        },
+    )
+    response.raise_for_status()
+    response = response.json()
+
+    is_superuser = admin_group_name in response.get("groups", [])
+    is_staff = admin_group_name in response.get("groups", [])
+
+    if user.is_superuser != is_superuser:
+        user.is_superuser = is_superuser
+        user.save()
+
+    if user.is_staff != is_staff:
+        user.is_staff = is_staff
+        user.save()
```

Comment:
I would suggest to batch the two `user.save()` calls so that we avoid potential weird states with the model being half updated, perhaps something like this:

```suggestion
    user_changed = False

    if user.is_superuser != is_superuser:
        user.is_superuser = is_superuser
        user_changed = True

    if user.is_staff != is_staff:
        user.is_staff = is_staff
        user_changed = True

    if user_changed:
        user.save()
```

Reply:
Same comment for `fetch_okta_openidconnect_permissions` below.

Reply:
Good point, done.